AutoRest/Generators/Extensions/Azure.Extensions/AzureExtensions.cs

```diff
@@ -77,7 +78,12 @@ public static void NormalizeAzureClientModel(ServiceClient serviceClient, Settin
             FlattenResourceProperties(serviceClient);
             FlattenRequestPayload(serviceClient, settings);
             AddPageableMethod(serviceClient);
-            AddLongRunningOperations(serviceClient);
+            CodeNamer codeNamer = default(CodeNamer);
+            if (settings != null && settings.CodeGenerator == "Azure.NodeJS")
```

Comment:
It's a good idea to have a dependency on code generator from extensions. Dependency should go the other way.